stubs/docker/docker/types/services.pyi

```diff
@@ -1,170 +1,185 @@
+        healthcheck: Healthcheck | None = None,
+        hosts: dict[str, str] | None = None,
+        dns_config: DNSConfig | None = None,
+        configs: list[ConfigReference] | None = None,
+        privileges: Privileges | None = None,
+        isolation: str | None = None,
+        init: bool | None = None,
+        cap_add: list[Incomplete] | None = None,
+        cap_drop: list[Incomplete] | None = None,
+        sysctls: dict[str, Incomplete] | None = None,
     ) -> None: ...
 
 class Mount(dict[str, Incomplete]):
     def __init__(
         self,
-        target,
-        source,
-        type: str = "volume",
+        target: str,
+        source: str,
+        type: Literal["bind", "volume", "tmpfs", "npipe"] = "volume",
         read_only: bool = False,
-        consistency: Incomplete | None = None,
-        propagation: Incomplete | None = None,
+        consistency: Literal["default", "consistent", "cached", "delegated"] | None = None,
+        propagation: str | None = None,
         no_copy: bool = False,
-        labels: Incomplete | None = None,
-        driver_config: Incomplete | None = None,
-        tmpfs_size: Incomplete | None = None,
-        tmpfs_mode: Incomplete | None = None,
+        labels: dict[Incomplete, Incomplete] | None = None,
+        driver_config: DriverConfig | None = None,
+        tmpfs_size: int | str | None = None,
+        tmpfs_mode: int | None = None,
     ) -> None: ...
     @classmethod
-    def parse_mount_string(cls, string): ...
+    def parse_mount_string(cls, string: str) -> Mount: ...
 
 class Resources(dict[str, Incomplete]):
     def __init__(
         self,
-        cpu_limit: Incomplete | None = None,
-        mem_limit: Incomplete | None = None,
-        cpu_reservation: Incomplete | None = None,
-        mem_reservation: Incomplete | None = None,
-        generic_resources: Incomplete | None = None,
+        cpu_limit: int | None = None,
+        mem_limit: int | None = None,
+        cpu_reservation: int | None = None,
+        mem_reservation: int | None = None,
+        generic_resources: dict[str, Incomplete] | list[str] | None = None,
     ) -> None: ...
 
 class UpdateConfig(dict[str, Incomplete]):
     def __init__(
         self,
         parallelism: int = 0,
-        delay: Incomplete | None = None,
-        failure_action: str = "continue",
-        monitor: Incomplete | None = None,
-        max_failure_ratio: Incomplete | None = None,
-        order: Incomplete | None = None,
+        delay: int | None = None,
+        failure_action: Literal["pause", "continue", "rollback"] = "continue",
+        monitor: int | None = None,
+        max_failure_ratio: float | None = None,
+        order: Literal["start-first", "stop-first"] | None = None,
     ) -> None: ...
 
 class RollbackConfig(UpdateConfig): ...
 
 class RestartConditionTypesEnum:
-    NONE: Incomplete
-    ON_FAILURE: Incomplete
-    ANY: Incomplete
+    NONE: str
+    ON_FAILURE: str
+    ANY: str
```

Comment:
Given this seems to be meant to emulate an enum, I feel like it would be accurate to type using `Final`. Especially if you expect those literals for parameters.
```suggestion
    NONE: Final = "none"
    ON_FAILURE: Final = "on-failure"
    ANY: Final = "any"
```